src/components/Holdable/Holdable.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import EventListener from 'react-event-listener';
+
+export default class Holdable extends React.Component {
+
+	static propTypes = {
+		children: React.PropTypes.element,
+		delay: React.PropTypes.number,
+		isDisabled: React.PropTypes.bool,
+		interval: React.PropTypes.number,
+		onHold: React.PropTypes.func
+	}
+
+	_timeoutId;
+	_intervalId;
+
+	static defaultProps = {
+		interval: 50,
+		delay: 300
+	};
+
+	componentWillReceiveProps(newProps) {
+		if (newProps.isDisabled) {
+			this.stop();
+		}
+	}
+
+	onMouseDown = () => {
+		const {interval, delay, onHold} = this.props;
+		this._timeoutId = setTimeout(() => {
+			this._intervalId = setInterval(() => {
+				onHold && onHold();
+			}, interval);
+
+			this.forceUpdate();
+		}, delay);
+	}
+
+	stop() {
+		if (this._timeoutId) {
+			clearTimeout(this._timeoutId);
```

Comment:
It's ok for now but actually both `clearTimeout` and `clearInverval` accept nulls so checking is redundant